src/util/xtuple.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+// Tuple-alike that allows implicit initialization from a brace-enclosed list.
+// This functionality is offered by std::tuple in C++17 and is an extension
+// in recent versions of libstdc++ and libc++.
```

Comment:
Have a clear justification in the comment about where and why this is needed.

Reply:
I can see why you would want to know, but I don't think this is the right place to explain it: this sort of documentation approach does not scale for utility classes, and introduces a sort of backwards dependency for documentation between use and definition.

The implicit brace initialization is used to simplify the generated mechanism code, and `tuple` is used instead of a POD struct so as to make generic access with `get<>` and the map find utility routines straightforward. Again, in C++17 the problem is solved at both ends: we can use structured binding to avoid using tuple, or we can use tuple with implicit construction from a brace-enclosed list.